### ISSUE TITLE
Add checking rule of Boolean type in SettingDef

### ISSUE THREAD

**Issue by saivirtue** [oharastream/ohara#2491]
For any definition of `SettingDef`, a boolean type should always have a value (**true** or **false**)

The following situation should throw an exception
```java
SettingDef
  .builder()
  .valueType(Type.BOOLEAN)
  .key("booleanKey")
  .build()
```
And the request
```json
{
  "booleanKey": null
}
```
or
 ```json
{
  "booleanKey": "fake"
}
```

**Comment by saivirtue** [comment]
Since this issue is an improvement for checker of SettingDef (which only implemented in 0.8), I will NOT backport this function to 0.7

/cc @oharastream/frontend @chia7712 @HarryChiang143

**Comment by chia7712** [comment]
> Since this issue is an improvement for checker of SettingDef (which only implemented in 0.8), I will NOT backport this function to 0.7

+1 to not backport :)
